packages/compiler/src/i18n/serializers/xmb.ts

```diff
@@ -48,6 +48,10 @@ const _DOCTYPE = `<!ELEMENT messagebundle (msg)*>
 <!ELEMENT ex (#PCDATA)>`;
 
 export class Xmb extends Serializer {
+  constructor(private readonly preservePlaceholders?: boolean) {
```

Comment:
```suggestion
  constructor(private readonly preservePlaceholders: boolean) {
```
(unless this needs to be optional for compatibility with external usages)

Reply:
This is exported from `@angular/compiler`, so I think it qualifies as public API and can't be changed right now. I left it optional, but did update all internal usages to explicitly pass this value in.